crates/rolldown/src/utils/renamer.rs

```diff
@@ -24,32 +42,39 @@ impl<'name> Renamer<'name> {
       used_canonical_names: RESERVED_KEYWORDS
         .iter()
         .chain(GLOBAL_OBJECTS.iter())
-        .map(|s| Cow::Owned(Rstr::new(s)))
+        .map(|s| (Cow::Owned(Rstr::new(s)), 0))
         .collect(),
     }
   }
 
   pub fn reserve(&mut self, name: Cow<'name, Rstr>) {
-    self.used_canonical_names.insert(name);
+    self.used_canonical_names.insert(name, 0);
   }
 
   pub fn add_top_level_symbol(&mut self, symbol_ref: SymbolRef) {
     let canonical_ref = self.symbols.par_canonical_ref_for(symbol_ref);
     let original_name: Cow<'_, Rstr> =
       Cow::Owned(self.symbols.get_original_name(canonical_ref).to_rstr());
-
     match self.canonical_names.entry(canonical_ref) {
-      std::collections::hash_map::Entry::Vacant(vacant) => {
-        let mut count = 0;
+      Entry::Vacant(vacant) => {
         let mut candidate_name = original_name.clone();
-        while self.used_canonical_names.contains(&candidate_name) {
-          count += 1;
-          candidate_name = Cow::Owned(format!("{original_name}${count}").into());
+        loop {
+          match self.used_canonical_names.entry(candidate_name.clone()) {
```

Comment:
```suggestion
          match self.used_canonical_names.entry(original_name.clone()) {
```

Should you use `original_name` name here?

Reply:
We should use candidate name here, you could ref the algorithm https://github.com/rolldown/rolldown/pull/1461/files#diff-c87caec33470b59eac43cd6ac05bef999e6a4fa010e716fd3e546971148f01d2R15,

Reply:
We need to find a fixed point

Reply:
If there's is another `a`, in the current `algorithm`, will it be renamed to `a$1$1$1`? If so, I would like to see if we could fix this, it's kind of ugly in the output.

If we don't care the how output looks, we could follow how parcel did which is renaming the vars to a meaningless hashed value.

Reply:
Fixed

Reply:
We can't use `original_name` here, it will lead a dead loop, you could see the case in https://github.com/rolldown/rolldown/pull/1461/files#diff-c87caec33470b59eac43cd6ac05bef999e6a4fa010e716fd3e546971148f01d2R15